csrc/ir/nodes.cpp

```diff
@@ -3425,12 +3425,6 @@ void TensorDomain::merge(int axis_o, int axis_i) {
       axis_o != axis_i,
       "Invalid merge detected, axes provided are the same axis.");
 
-  if (axis_o > axis_i) {
```

Comment:
Just in case, can you run all tests and benchmarks with:

```
 if (axis_o > axis_i) {
   NVF_ERROR(false);
 }
```

to make sure nothing relies on this behavior?

Reply:
looks like there's only `FusionMappingRelation_CUDA` that happens to hit this with a `tv->merge(-1)`.

Reply:
similarly there's the python test that does a `merge(-1)` on the python schedule tests.

Reply:
so except those places where we have test where we are merging(-1). I'm not seeing any other use case with this intentionally in our scheduler.

Surprisingly this one is good to go :exploding_head: